planner/core/planbuilder.go

```diff
@@ -690,43 +657,170 @@ func (b *PlanBuilder) buildAdmin(ctx context.Context, as *ast.AdminStmt) (Plan,
+				colsMap[c.ID] = struct{}{}
+				if mysql.HasPriKeyFlag(c.Flag) {
+					pkOffset = len(tblColumns) - 1
+				}
+			}
+		}
+	}
+	if !tbl.Meta().PKIsHandle || pkOffset == -1 {
+		tblColumns = append(tblColumns, model.NewExtraHandleColInfo())
+		handleCol := &expression.Column{
+			DBName:   dbName,
+			TblName:  tblInfo.Name,
+			ColName:  model.ExtraHandleName,
+			RetType:  types.NewFieldType(mysql.TypeLonglong),
+			UniqueID: b.ctx.GetSessionVars().AllocPlanColumnID(),
+			ID:       model.ExtraHandleID,
+		}
+		tblSchema.Append(handleCol)
+		pkOffset = len(tblColumns) - 1
+	}
 
-			expr, _, err := b.rewrite(ctx, column.GeneratedExpr, mockTablePlan, nil, true)
+	is := PhysicalIndexScan{
+		Table:            tblInfo,
+		TableAsName:      &tblInfo.Name,
+		DBName:           dbName,
+		Columns:          columns,
+		Index:            idx,
+		dataSourceSchema: schema,
+		Ranges:           ranger.FullRange(),
+		KeepOrder:        false,
+		GenExprs:         genExprs,
+	}.Init(b.ctx)
+	is.stats = property.NewSimpleStats(0)
+	cop := &copTask{indexPlan: is}
+	// It's double read case.
+	ts := PhysicalTableScan{Columns: tblColumns, Table: is.Table}.Init(b.ctx)
+	ts.SetSchema(tblSchema)
+	cop.tablePlan = ts
+	ts.HandleIdx = pkOffset
+	is.initSchema(id, idx, true)
+	t := finishCopTask(b.ctx, cop)
+	rootT := t.(*rootTask)
+	return rootT.p, nil
+}
+
+func (b *PlanBuilder) buildPhysicalIndexLookUpReaders(ctx context.Context, dbName model.CIStr, tbl table.Table) ([]Plan, []table.Index, error) {
+	tblInfo := tbl.Meta()
+	// get index information
+	indices := make([]table.Index, 0, len(tblInfo.Indices))
+	indexLookUpReaders := make([]Plan, 0, len(tblInfo.Indices))
+	for i, idx := range tbl.Indices() {
+		idxInfo := idx.Meta()
+		if idxInfo.State != model.StatePublic {
+			logutil.Logger(context.Background()).Info("build physical index lookup reader, the index isn't public",
+				zap.String("index", idxInfo.Name.O), zap.Stringer("state", idxInfo.State), zap.String("table", tblInfo.Name.O))
+		} else {
+			indices = append(indices, idx)
+			reader, err := b.buildPhysicalIndexLookUpReader(ctx, dbName, tbl, idxInfo, i)
 			if err != nil {
-				return nil, err
+				return nil, nil, err
 			}
-			expr = expression.BuildCastFunction(b.ctx, expr, colExpr.GetType())
-			p.GenExprs[model.TableColumnID{TableID: tableInfo.ID, ColumnID: column.ColumnInfo.ID}] = expr
+			indexLookUpReaders = append(indexLookUpReaders, reader)
 		}
 	}
+	if len(indexLookUpReaders) == 0 {
+		return nil, nil, nil
+	}
+	return indexLookUpReaders, indices, nil
+}
+
+func (b *PlanBuilder) buildAdminCheckTable(ctx context.Context, as *ast.AdminStmt) (*CheckTable, error) {
+	tbl := as.Tables[0]
+	p := &CheckTable{
+		DBName:  tbl.Schema.O,
+		TblInfo: tbl.TableInfo,
+	}
+
+	mockTablePlan := LogicalTableDual{}.Init(b.ctx)
+	tableInfo := as.Tables[0].TableInfo
+	schema := expression.TableInfo2SchemaWithDBName(b.ctx, tbl.Schema, tableInfo)
+	mockTablePlan.SetSchema(schema)
```

Comment:
It's seems like we have defined a mockTablePlan here without use. 
Similar thing in `buildPhysicalIndexLookUpReader`—— downstream of the call chain.